### ISSUE TITLE
"Editing a clip can move other clips" is not honored by smart clip resize or by time stretching the clip

### ISSUE THREAD

**Issue by petersampsonaudacity** [audacity/audacity#5161]
### Bug description

The **Tracks Behaviors** preference "**Editing a clip can move other clips**" is not honored by smart clip resize or by time stretching the clip.

Note that the default setting for this pref is "off" so normally the other clips would not move - and setting this is a direct choice by the user that they want their subsequent clips in the track that is being edited to move.

### Steps to reproduce

1. In Tracks Behaviors prefs set **Editing a clip can move other clips** to be on _(non-default setting)_
2. Generate a 30 second chirp
3. zoom out 2 levels
4. set cursor at T = 1min
5. Generate a 2nd 30 second chirp
6. In the first clip select from 10-20 seconds
7. Ctrl+K to delete the selection
8. Observe: the second clip moves forward 10 seconds to start at T=50 secs
9. Shrink the first clip from the left wither with a smart clip drag of a time-stretch drag
10. Observe: the second clip does **_NOT_** move

### Expected behavior

When **Editing a clip can move other clips** is turned "on" then editing a prior clip in a track by either  smart clip resizing of by time-stretching should cause subsequent clips to the right of the edited clip to be moved accordingly.

### Actual behavior

Step 6
![image](https://github.com/audacity/audacity/assets/50205984/de51fbbb-9288-49ef-b66c-0cdeb4ea6d5b)

**Step 8**
![image](https://github.com/audacity/audacity/assets/50205984/5d8ff8f5-8bf1-4e5d-869d-0fd2fd00472c)

**step 10**
![image](https://github.com/audacity/audacity/assets/50205984/154e6772-c9e6-48ec-9c05-8ee3075a458a)

### Audacity Version

current master / alpha / nightly build

### Operating system

Windows

### Additional context

3.3.3 Exhibits this behavior with smart-clip right-edge dragging

3.4.0 alpha exhibits this with smart-clip right-edge dragging and with clip time-stretching

flagging @saintmatthieu @LWinterberg @crs

**Comment by petersampsonaudacity** [comment]
Also when this preference is on stretch a clip up to and beyond the next clip on the track should move that clip and all subsequent clips on the track.

This is what happens if you paste in where there is insufficient room between clips when this pref is "on"

**Comment by LWinterberg** [comment]
mouse drags don't ripple-edit other clips generally. Like, moving a clip doesn't move around other clips either.

**Comment by petersampsonaudacity** [comment]
A good use case for this would be the preservation of lengths of silence between clips

**Comment by LWinterberg** [comment]
Preserving the distance between clips is a goal, but what's the use case?

**Comment by petersampsonaudacity** [comment]
Use Case
Podcast user has several clips in  track separated by silences of a specific length
User the stretches a music clip where there are further clips to the right
user wishes to maintain the specific length of the inter-clip silence

**Comment by saintmatthieu** [comment]
Sounds like a valid use case to me.
Discussing it this morning with Peter, he recounted that the "edit clips can move" setting was turned off (because that's a nightmare for musicians), but the "There is not enough room" error annoyed users and it was restored in the preferences as a checkbox.
I suppose this was a feeble, provisional solution, waiting for overlapping clips.
So when we _do_ have overlapping clips, musicians can have the setting off and not be annoyed, and only podcasters would have it checked, for a good reason. At that stage I would indeed see this request as a sensible one.

**Comment by LWinterberg** [comment]
> Discussing it this morning with Peter, he recounted that the "edit clips can move" setting was turned off (because that's a nightmare for musicians), but the "There is not enough room" error annoyed users and it was restored in the preferences as a checkbox.

I don't think that option got removed, rather, it just got disabled by default for new installs. The thing that was removed and reinstated later are "the scissors" in the form of a cut/copy/paste toolbar.

Regarding the feature request, it's a very rare feature. Premiere and Davinci resolve have it, but I don't think I've seen an audio editor or DAW with this feature. Reaper, Hindenburg and Audition, which probably are the must likely destinations for podcasters outside Audacity, don't have it anyway. 

This indicates to me that while ripple delete/cut/insert are very well known features, present in virtually everything with a timeline, ripple trim is a pretty obscure function. I'd be cautious having ripple trim on a shared toggle with ripple cut.

**Comment by petersampsonaudacity** [comment]
Testing on W10 with @saintmatthieu latest 4 of 6 branch build:  audacity-win-3.4.0-alpha-20231005+7737b4b-x64

This remains the case in this branch build.

**Comment by saintmatthieu** [comment]
Sorry if there was a misunderstanding, Peter: there is no intention to address this in 4-of-6.

**Comment by petersampsonaudacity** [comment]
This was more in the way of a "bump" Matthieu - just retesting an issue that I logged earlier (and one I think that should be fixed)

**Comment by LWinterberg** [comment]
As laid out earlier, ripple trim is an extremely rare feature to begin with only found in video editing software otherwise afaict, and while we *may* add it at some point in the future, I wouldn't want to introduce it now, either through expanding the scope of this preference or through real ripple editing.
